boards/arm/nrf52/arduino-nano-33ble/scripts/flash_config.ld

```diff
@@ -0,0 +1,106 @@
+/****************************************************************************
+ * boards/arm/nrf52/arduino-nano-33ble/scripts/flash_config.ld
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+MEMORY
+{
+    progmem (rx)  : ORIGIN = 0x00010000, LENGTH = 960K
```

Comment:
is there any built-in bootloader? if so, maybe we should configure the bootloader presence with with Kconfig ?

Reply:
There is. What would you want to predicate on that config option? Should I just use ARCH_HAVE_BOOTLOADER, or create a new one?

Reply:
If this is arduino factory programmed bootlaoder, then it is best to place`ARDUINONANO33BLE_ARDUINO_BOOTLOADER` (or sth like that) option in `arduino-nano-33ble/Kconfig`

Reply:
OK will do. What should I do differently if that option is not set? Go ahead and start the text at 0 instead of 0x10000? FYI I have no intention of blowing away my bootloader to test that. :D

Reply:
Yes, let's start from 0.  This will allow the user to program custom bootloaders, for example based on NuttX and mcuboot :)